### ISSUE TITLE
[0.4.5] eqeqeq ignores the null:ignore and null:never options

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1274]
## Summary

The explicit null exception is ignored: x == null and null != x are reported with null:ignore. With null:never, loose comparisons are also incorrectly reported while x === null is missed. The always option and non-null comparison controls behave as expected.

Verified with @utoo/lint 0.4.5, ESLint 8.57.1, @typescript-eslint/parser and eslint-plugin 6.21.0, eslint-plugin-react 7.37.5, Node.js 22.23.1 on macOS arm64. Each example enables only the indicated rule, with no parsing failures or suppression comments.

## Setup

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.5 eslint@8.57.1 @typescript-eslint/parser@6.21.0 @typescript-eslint/eslint-plugin@6.21.0 eslint-plugin-react@7.37.5 typescript@5.3.3
```

`tsconfig.json`:

```json
{"compilerOptions":{"strict":true,"target":"ES2022","jsx":"react"},"include":["example.tsx"]}
```

Run each case separately, replacing example.tsx and both config files with the corresponding contents below:

```sh
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
```

## ignore-eq-null

`example.tsx`:

```tsx
export function example(x){return x==null;}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [],
  "rules": {
    "eqeqeq": [
      "error",
      "always",
      {
        "null": "ignore"
      }
    ]
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "eqeqeq": [
      "error",
      "always",
      {
        "null": "ignore"
      }
    ]
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
}
```

ESLint: 0 diagnostic(s). utlint: 1 diagnostic(s).

eslint messages:

```text
(none)
```

utlint messages:

```text
Use strict equality operators.
```

## ignore-eq-reversed

`example.tsx`:

```tsx
export function example(x){return null!=x;}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [],
  "rules": {
    "eqeqeq": [
      "error",
      "always",
      {
        "null": "ignore"
      }
    ]
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "eqeqeq": [
      "error",
      "always",
      {
        "null": "ignore"
      }
    ]
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
}
```

ESLint: 0 diagnostic(s). utlint: 1 diagnostic(s).

eslint messages:

```text
(none)
```

utlint messages:

```text
Use strict equality operators.
```

## ignore-eq-number

`example.tsx`:

```tsx
export function example(x){return x==1;}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [],
  "rules": {
    "eqeqeq": [
      "error",
      "always",
      {
        "null": "ignore"
      }
    ]
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "eqeqeq": [
      "error",
      "always",
      {
        "null": "ignore"
      }
    ]
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
}
```

ESLint: 1 diagnostic(s). utlint: 1 diagnostic(s).

eslint messages:

```text
Expected '===' and instead saw '=='.
```

utlint messages:

```text
Use strict equality operators.
```

## ignore-strict-null

`example.tsx`:

```tsx
export function example(x){return x===null;}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [],
  "rules": {
    "eqeqeq": [
      "error",
      "always",
      {
        "null": "ignore"
      }
    ]
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "eqeqeq": [
      "error",
      "always",
      {
        "null": "ignore"
      }
    ]
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
}
```

ESLint: 0 diagnostic(s). utlint: 0 diagnostic(s).

eslint messages:

```text
(none)
```

utlint messages:

```text
(none)
```

## always-eq-null

`example.tsx`:

```tsx
export function example(x){return x==null;}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [],
  "rules": {
    "eqeqeq": [
      "error",
      "always",
      {
        "null": "always"
      }
    ]
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "eqeqeq": [
      "error",
      "always",
      {
        "null": "always"
      }
    ]
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
}
```

ESLint: 1 diagnostic(s). utlint: 1 diagnostic(s).

eslint messages:

```text
Expected '===' and instead saw '=='.
```

utlint messages:

```text
Use strict equality operators.
```

## never-eq-null

`example.tsx`:

```tsx
export function example(x){return x==null;}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [],
  "rules": {
    "eqeqeq": [
      "error",
      "always",
      {
        "null": "never"
      }
    ]
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "eqeqeq": [
      "error",
      "always",
      {
        "null": "never"
      }
    ]
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
}
```

ESLint: 0 diagnostic(s). utlint: 1 diagnostic(s).

eslint messages:

```text
(none)
```

utlint messages:

```text
Use strict equality operators.
```

## never-strict-null

`example.tsx`:

```tsx
export function example(x){return x===null;}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [],
  "rules": {
    "eqeqeq": [
      "error",
      "always",
      {
        "null": "never"
      }
    ]
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "eqeqeq": [
      "error",
      "always",
      {
        "null": "never"
      }
    ]
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
}
```

ESLint: 1 diagnostic(s). utlint: 0 diagnostic(s).

eslint messages:

```text
Expected '==' and instead saw '==='.
```

utlint messages:

```text
(none)
```
